test/inc/zoo/variant.h

```diff
@@ -9,6 +10,25 @@
 
 namespace zoo {
 
+template<typename... Ts>
+struct Variant;
+
+} // namespace zoo
+
+namespace std {
+
+template<typename... Ts>
+struct is_trivially_destructible<zoo::Variant<Ts...>> :
+    integral_constant<
```

Comment:
Nope, specializing `is_trivially_destructible` is not allowed, besides, it does not work either

Reply:
Don't know about the magic requirement. The test seems to pass fine.

Reply:
I understand your PR better now.  This specialization of `is_trivially_destructible` is not needed, the static assertions succeed because `Variant` has the implicit default destructor, whose triviality is set by its base class `DestroyInvoker` trivial destructibility.  Because of the partial specialization, `DestroyInvoker` is or not trivially destructible correctly.  But still this illegal specialization of `std::is_trivially_destructible` (illegal because it is not one of the exceptions to the rule of not putting something in the `std` namespace) is redundant

Reply:
Please check this link:
https://godbolt.org/z/W842oi
You'll see that specializing `is_trivially_destructible` has no effect on derived types